docs/pipeline.rst

```diff
@@ -51,23 +54,42 @@ Current pipeline
 """"""""""""""""
 
 Subgraphs that do not contribute to the pipeline output are automatically pruned.
-If an operator has side effects (e.g. PythonFunction operator family), it cannot be invoked
+If an operator has side effects (e.g. ``PythonFunction`` operator family), it cannot be invoked
 without setting the current pipeline. Current pipeline is set implicitly when the graph is
-defined inside derived pipeline's `define_graph` method. Otherwise, it can be set using context
-manager (`with` statement)::
+defined inside derived pipelines' :meth:`Pipeline.define_graph` method.
+Otherwise, it can be set using context manager (``with`` statement)::
 
-    pipe = dali.pipeline.Pipeline(batch_size = N, num_threads = 3, device_id = 0)
+    pipe = dali.Pipeline(batch_size=N, num_threads=3, device_id=0)
     with pipe:
-        src = dali.ops.ExternalSource(my_source, num_outputs = 2)
+        src = dali.ops.ExternalSource(my_source, num_outputs=2)
         a, b = src()
         pipe.set_outputs(a, b)
 
+When creating a pipeline with :meth:`pipeline_def`, the function which defines the pipeline is
+executed within the scope of the newly created pipeline. The following example is equivalent
+to the previous one::
+
+    @dali.pipeline_def(batch_size=N, num_threads=3, device_id=0)
+    def my_pipe(my_source):
+        return dali.fn.external_source(my_source, num_outputs=2)
+
+    pipe = my_pipe(my_source)
+
+And the decorated function will return ``Pipeline`` object.
+You can find more info about decorator syntax and how to use it in the
+:meth:`pipeline` documentation below.
```

Comment:
This paragraph shifts attention from the topic at hand (it's in the "Current pipeline" section). I can understand your excitement with this new and **awesome** feature, but this is not the place to advertise it. You can elaborate on it in the first two sections.